protocol/app/ante.go

```diff
@@ -124,15 +124,21 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 		),
 		sigVerification: accountplusante.NewCircuitBreakerDecorator(
 			options.Codec,
-			accountplusante.NewAuthenticatorDecorator(
-				options.Codec,
-				options.AccountplusKeeper,
-				options.AccountKeeper,
-				options.SignModeHandler,
+			sdk.ChainAnteDecorators(
+				accountplusante.NewAuthenticatorDecorator(
+					options.Codec,
+					options.AccountplusKeeper,
+					options.AccountKeeper,
+					options.SignModeHandler,
+				),
 			),
-			customante.NewSigVerificationDecorator(
-				options.AccountKeeper,
-				options.SignModeHandler,
+			sdk.ChainAnteDecorators(
```

Comment:
For my understanding why was it necessary to move `SetPubKey` and `SigGas` under this chain and thus not applicable to smart account flow?

Reply:
SetPubKey expects the signer to match the public key ([link](https://github.com/dydxprotocol/cosmos-sdk/blob/dydx-fork-v0.50.5/x/auth/ante/sigverify.go#L92-L96)), and I was getting some errors here. 

After moving `SetPubKey`, I was getting some `public key = nil` errors in `SigGas` [here](https://github.com/dydxprotocol/cosmos-sdk/blob/dydx-fork-v0.50.5/x/auth/ante/sigverify.go#L455) because public key is not set. So moved this as well.

This is also how osmosis does it ([link](https://github.com/osmosis-labs/osmosis/blob/2d938d0d31ea35c11286a46cc30f9a2e03997332/app/ante.go#L73))